### PR TITLE
[WIP] Polish code & add consistency between the R4 and DSi theme

### PR DESCRIPTION
**Pull request by NightYoshi370**
#### What's changed?

1. (SETTINGS) Now instead of making multiple functions per SD card themes, just have one for all themes, with the folder and "array" in the function arguments
2. Remove old comments
3. (R4) Port over the Hide ROMs feature.
4. The delete and hide menu is now a "ROM Management" menu, as it makes much more sense grammatically.
5. Just overall more consistency between the r4 and DSi theme
6. Overall code cleanup

#### Where have you tested it?

I have not tested the R4 Hide ROMs feature & the Settings Theme change. However, the rest should not require testing.

#### Pull Request status
- [x]  This PR has been tested using the provided devkitPro, devkitARM, and EasyGL2D.  

_(Do not edit after this point)_
- [ ]  This PR is fully documented.
- [ ]  This PR has been tested before sending.
- [ ]  This PR follows C style and convention.

### CODE REVIEW

**Comment by RocketRobz** [comment]
Travis throws these errors:
```
/home/travis/build/RocketRobz/TWiLightMenu/settings/arm9/source/main.cpp: In function 'int main(int, char**)':
/home/travis/build/RocketRobz/TWiLightMenu/settings/arm9/source/main.cpp:335:36: error: invalid initialization of non-const reference of type 'std::vector<std::__cxx11::basic_string<char> >&' from an rvalue of type 'std::vector<std::__cxx11::basic_string<char> >*'
  loadThemeList(AK_THEME_DIRECTORY, &akThemeList);
                                    ^~~~~~~~~~~~
/home/travis/build/RocketRobz/TWiLightMenu/settings/arm9/source/main.cpp:201:6: note: in passing argument 2 of 'void loadThemeList(const char*, std::vector<std::__cxx11::basic_string<char> >&)'
 void loadThemeList(const char* directory, std::vector<std::string>& list)
      ^~~~~~~~~~~~~
/home/travis/build/RocketRobz/TWiLightMenu/settings/arm9/source/main.cpp:336:36: error: invalid initialization of non-const reference of type 'std::vector<std::__cxx11::basic_string<char> >&' from an rvalue of type 'std::vector<std::__cxx11::basic_string<char> >*'
  loadThemeList(R4_THEME_DIRECTORY, &r4ThemeList);
                                    ^~~~~~~~~~~~
/home/travis/build/RocketRobz/TWiLightMenu/settings/arm9/source/main.cpp:201:6: note: in passing argument 2 of 'void loadThemeList(const char*, std::vector<std::__cxx11::basic_string<char> >&)'
 void loadThemeList(const char* directory, std::vector<std::string>& list)
      ^~~~~~~~~~~~~
```

**Comment by NightYoshi370** [comment]
I saw, and to that I say the following: 
image

I pinged you on Rocket.Chat asking for help

File: settings/arm9/source/main.cpp [hunk 1]
```diff
@@ -198,12 +198,13 @@ void loadROMselect()
 	}
 }
 
-void loadThemeList(const char* directory, std::vector<std::string>& list)
```
Comment (arraystock) [inline settings/arm9/source/main.cpp]: @NightYoshi370 If you change this to ` void loadThemeList(const char* directory, std::vector<std::string>* list)`

File: settings/arm9/source/main.cpp [hunk 2]
```diff
@@ -213,7 +214,28 @@ void loadThemeList(const char* directory, std::vector<std::string>& list)
 			themeDir = ent->d_name;
 			if (themeDir == ".." || themeDir == "..." || themeDir == ".") continue;
 
-			list.emplace_back(themeDir);
```
Comment (arraystock) [inline settings/arm9/source/main.cpp]: and use `list->emplace_back(themeDir);` Then this properly compiles (and seems to work on my end).

**Pull request merged**